presto-product-tests/src/main/java/com/facebook/presto/tests/AlterTableTests.java

```diff
@@ -32,20 +33,70 @@
     @Test(groups = {ALTER_TABLE, SMOKE})
     public void renameTable()
     {
-        String tableName = "to_be_renamed_table_name";
+        String tableName = "to_be_renamed";
         String renamedTableName = "renamed_table_name";
+
         query(format("DROP TABLE IF EXISTS %s", tableName));
         query(format("DROP TABLE IF EXISTS %s", renamedTableName));
 
-        query(format("CREATE TABLE %s AS SELECT * FROM nation", tableName));
-        assertThat(query(format("ALTER TABLE %s RENAME TO %s", tableName, renamedTableName), UPDATE))
-                .hasRowsCount(1);
+        try {
+            query(format("CREATE TABLE %s AS SELECT * FROM nation", tableName));
+
+            assertThat(query(format("ALTER TABLE %s RENAME TO %s", tableName, renamedTableName), UPDATE))
+                    .hasRowsCount(1);
+
+            assertThat(query(format("SELECT * FROM %s", renamedTableName)))
+                    .hasRowsCount(25);
+
+            // rename back to original name
+            assertThat(query(format("ALTER TABLE %s RENAME TO %s", renamedTableName, tableName), UPDATE))
+                    .hasRowsCount(1);
+        }
+        finally {
+            query(format("DROP TABLE %s", tableName));
```

Comment:
Doing this in a finally block has the issue that if an exception is thrown by the tests and the `DROP TABLE` fails, the exception from the finally block will be the one thrown, hiding the reason for the test failure.

We handle this in `AbstractTestHiveClient` by having a cleanup method `dropTable` which just logs the failure. Can you send a new PR to add something similar for product tests? Maybe a generic `cleanup` method that takes a lambda and just logs the exception?